profiles: apparmor: remove unused fields

cyphar opened the pull request:
The template was referring to an apparmor profile that we don't ship, we
can just nix the lines from the profile and remove the fields in the
Data struct.

Fixes #20584

![cute kitty](http://farm4.staticflickr.com/3260/2789604490_066dd1c786_b.jpg)

Signed-off-by: Aleksa Sarai asarai@suse.de

/cc @icecrime @jfrazelle

jfrazelle commented:
So the peer-profile is referring to the apparmor profile not the binary
path, so this is cases where the daemon is running with the /usr/bin/docker
profile it's confusing because depending on the daemon profile being
installed it could mean different things, but it's not technically the
daemon path, since it's reliant on a profile we are not even shipping, I
say we nix the corresponding lines from the docker-default profile entirely
since it's not even applicable without the daemon profile

On Sunday, March 13, 2016, Aleksa Sarai <notifications@github.com> wrote:

> In order to deal with cases where the docker binary is not in $PATH,
> read the full path of the Docker binary from /proc/self/exe (which is
> guaranteed to always be an absolute path and always be correct for the
> process -- unless the file has been unlinked in which case there's not
> much we can do).
>
> Previously the version field for the template was not filled, resulting
> in all version-based checks failing. Fix this by actually filling the
> field.
>
> Fixes #20584 <https://github.com/docker/docker/issues/20584>
>
> [image: cute kitty]
> <https://camo.githubusercontent.com/420e8b4262439d47560d46d6027cdbb0b0e0df24/687474703a2f2f6661726d342e737461746963666c69636b722e636f6d2f333236302f323738393630343439305f303636646431633738365f622e6a7067>
>
> Signed-off-by: Aleksa Sarai asarai@suse.de
> <javascript:_e(%7B%7D,'cvml','asarai@suse.de');>
>
> /cc @icecrime <https://github.com/icecrime> @jfrazelle
> <https://github.com/jfrazelle>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/docker/docker/pull/21176
> Commit Summary
>
>    - profiles: apparmor: set ExecPath to $(readlink /proc/self/exe)
>    - profiles: apparmor: actually fill .Version
>
> File Changes
>
>    - *M* profiles/apparmor/apparmor.go
>    <https://github.com/docker/docker/pull/21176/files#diff-0> (6)
>
> Patch Links:
>
>    - https://github.com/docker/docker/pull/21176.patch
>    - https://github.com/docker/docker/pull/21176.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/21176>.
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

cyphar commented:
@jfrazelle Wait, so both of the version-based `{{if}}`s are actually not valid (in which case we can nix `Version` as well as `ExecPath` and the `peer-profile`).

jfrazelle commented:
Yup

On Sunday, March 13, 2016, Aleksa Sarai <notifications@github.com> wrote:

> @jfrazelle <https://github.com/jfrazelle> Wait, so both of the
> version-based {{if}}s are actually not valid (in which case we can nix
> Version as well as ExecPath and the peer-profile).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/21176#issuecomment-196160932>.
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

cyphar commented:
@jfrazelle There, I've updated this PR.

jfrazelle commented on profiles/apparmor/template.go:
```diff
@@ -37,14 +37,5 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
-
-{{if ge .Version 208095}}
-  # suppress ptrace denials when using 'docker ps' or using 'ps' inside a container
-  ptrace (trace,read) peer=docker-default,
```
ok this peer we need sorry, because its going back to itself

cyphar replied:
Haha, no problem. I'll need to re-add the version checks (the old version didn't set the version so this was never enabled).

cyphar committed:
profiles: apparmor: actually calculate version

In order to check that we can have the `ptrace` rule, we need to
actually calculate the version of apparmor_parser.

Signed-off-by: Aleksa Sarai <asarai@suse.de>

cyphar committed:
profiles: apparmor: remove unused fields

ExecPath isn't used by anything, and the signal apparmor rule isn't used
because it refers to a peer that we don't ship.

Signed-off-by: Aleksa Sarai <asarai@suse.de>

cyphar commented:
@jfrazelle There, fixed.

thaJeztah commented:
WindowsTP4 failing on https://github.com/docker/docker/issues/21247 again (flaky test)

cyphar commented:
The `win2lin` failure is because the server has run out of memory (or it's a `containerd` issue?).

cyphar commented:
It's all :green_heart:.

/ping @jfrazelle

jfrazelle commented:
LGTM!
